core/modules/user/user.module

```diff
@@ -1417,7 +1417,7 @@ function user_get_user_login_paths() {
     $patterns['login'] = array();
     $patterns['non_login'] = array();
     foreach ($paths as $path => $enabled) {
-      if ($enabled) {
+      if ($enabled && backdrop_valid_path($path)) {
```

Comment:
Apparently we can't use backdrop_valid_path() here, as the hook uses asterisk placeholders, but this function can't deal with those.